entity-service/src/main/kotlin/com/egm/stellio/entity/web/EntityHandler.kt

```diff
@@ -163,33 +148,34 @@ class EntityHandler(
         @PathVariable entityId: String,
         @RequestParam params: MultiValueMap<String, String>
     ): ResponseEntity<*> {
-        val includeSysAttrs = params.getOrDefault(QUERY_PARAM_OPTIONS, emptyList())
-            .contains(QUERY_PARAM_OPTIONS_SYSATTRS_VALUE)
-        val useSimplifiedRepresentation = params.getOrDefault(QUERY_PARAM_OPTIONS, emptyList())
-            .contains(QUERY_PARAM_OPTIONS_KEYVALUES_VALUE)
         val contextLink = getContextFromLinkHeaderOrDefault(httpHeaders)
         val mediaType = getApplicableMediaType(httpHeaders)
         val sub = getSubFromSecurityContext()
 
+        val queryParams = parseAndCheckQueryParams(
+            Pair(applicationProperties.pagination.limitDefault, applicationProperties.pagination.limitMax),
+            params,
+            contextLink
+        )
+
         if (!entityService.exists(entityId.toUri()))
             throw ResourceNotFoundException(entityNotFoundMessage(entityId))
         if (!authorizationService.userCanReadEntity(entityId.toUri(), sub))
             throw AccessDeniedException("User forbidden read access to entity $entityId")
 
-        val jsonLdEntity = entityService.getFullEntityById(entityId.toUri(), includeSysAttrs)
+        val jsonLdEntity = entityService.getFullEntityById(entityId.toUri(), queryParams.includeSysAttrs)
 
-        val expandedAttrs = parseAndExpandRequestParameter(params.getFirst("attrs"), contextLink)
-        if (jsonLdEntity.containsAnyOf(expandedAttrs)) {
+        if (jsonLdEntity.containsAnyOf(queryParams.expandedAttrs)) {
             val filteredJsonLdEntity = JsonLdEntity(
-                JsonLdUtils.filterJsonLdEntityOnAttributes(jsonLdEntity, expandedAttrs),
+                JsonLdUtils.filterJsonLdEntityOnAttributes(jsonLdEntity, queryParams.expandedAttrs),
```

Comment:
this one also will no longer be expanded

Reply:
I expended it